babel/dates.py

```diff
@@ -1195,14 +1195,17 @@ class ParseError(ValueError):
 def parse_date(
     string: str,
     locale: Locale | str | None = LC_TIME,
-    format: _PredefinedTimeFormat = 'medium',
+    format: _PredefinedTimeFormat | str = 'medium',
 ) -> datetime.date:
     """Parse a date from a string.
 
     This function first tries to interpret the string as ISO-8601
     date format, then uses the date format for the locale as a hint to
     determine the order in which the date fields appear in the string.
 
+    If an explicit format is provided, the function will use it to parse
+    the date instead.
```

Comment:
This should note that the ISO-8601 attempt isn't done when you use an explicit format. Might need to rephrase the whole thing to make that flow nicely.

Reply:
I changed it up a bit so that it hopefully reads better now. Also added an example with an explicit `format=...`.